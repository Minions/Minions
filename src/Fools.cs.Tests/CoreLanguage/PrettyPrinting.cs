// PrettyPrinting.cs
// 
// Copyright 2012 The Minions Project (http:/github.com/Minions).
// All rights reserved. Usage as permitted by the LICENSE.txt file for this project.

using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using Fools.cs.Utilities;
using Fools.cs.builtins;
using NUnit.Framework;

namespace Fools.cs.Tests.CoreLanguage
{
	[TestFixture, UseReporter(typeof (DiffReporter))]
	public class PrettyPrinting
	{
		[Test]
		public void should_pretty_print_objects_with_action_and_func_properties_correctly()
		{
			var result = new ClassWithManyProperties().pretty_print();
			Approvals.Verify(result + "\r\n");
		}

		[UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
		private class ClassWithManyProperties
		{
			public int integer { get; private set; }
			public Action named_action { get; set; }
			public Action<double, string> lambda_action { get; set; }
			public Func<int, string, int> named_function { get; set; }

			public ClassWithManyProperties()
			{
				integer = 3;
				named_action = pass;
				lambda_action = (a, b) => { };
				named_function = big_fun;
			}

			private static int big_fun(int first_arg, string second_arg)
			{
				return 4;
			}

			private static void pass() { }
		}
	}
}
