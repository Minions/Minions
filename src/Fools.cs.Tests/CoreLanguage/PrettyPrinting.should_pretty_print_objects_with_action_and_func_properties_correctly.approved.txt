{
  "$type": "Fools.cs.Tests.CoreLanguage.PrettyPrinting+ClassWithManyProperties, Fools.cs.Tests",
  "integer": 3,
  "named_action": "Fools.cs.builtins.Missions:Void pass()",
  "lambda_action": "Fools.cs.Tests.CoreLanguage.PrettyPrinting+ClassWithManyProperties:Void <.ctor>b__0(Double, System.String)",
  "named_function": "Fools.cs.Tests.CoreLanguage.PrettyPrinting+ClassWithManyProperties:Int32 big_fun(Int32, System.String)"
}
